cassandane/tiny-tests/JMAPCalendars/calendarevent-encode-imip-uri

```diff
@@ -69,6 +71,9 @@ EOF
     my ($imipnotif) = grep { $_->{METHOD} eq 'imip' } @$data;
     my $payload = decode_json($imipnotif->{MESSAGE});
     $self->assert_str_equals('plus+uri@example.com', $payload->{recipient});
+    my $expect_id = "E-$uid";
```

Comment:
Please replace the hardcoded event id strings with `$self->encode_eventid($uid, $recurid)`. The recurid argument is optional.

Reply:
Oh, great tip, thanks!

Reply:
```
    my $expect_id = $self->encode_eventid($uid);
    xlog "XXX expect id $uid => $expect_id";
```

```
[4318] =====> Cyrus::JMAPCalendars::test_calendarevent_encode_imip_uri[75] XXX expect id event1uid => ERB-event1uid-Q2Fzc2FuZGFuZTo6Q3lydXM6OkpNQVBDYWxlbmRhcnM9SEFTSCgweDU2MWRlMjhhNjczOCk
```

Weird, `encode_eventid($uid)` is adding a load of (base64?) junk as a recurid when the recurid argument is omitted.

It's strange because I see lots of other tests using `encode_eventid($uid)` without an argument, and they're not failing.  I guess maybe they're comparing eventids where both were generated by Cassandane, so the difference from Cyrus's encoder isn't noticed?  Hmm.

Reply:
Oh I see what's wrong, it's not an object method, it's just a normal function.  So I shouldn't be calling it with `$self->` :)